modules/gapi/src/streaming/onevpl/cfg_param_device_selector.cpp

```diff
@@ -153,7 +180,78 @@ CfgParamDeviceSelector::CfgParamDeviceSelector(const CfgParams& cfg_params) :
             break;
         }
         case MFX_IMPL_VIA_VAAPI : {
-            GAPI_LOG_WARNING(nullptr, "TODO MFX_IMPL_VIA_VAAPI falls back to CPU case")
+#ifdef __linux__
+#if defined(HAVE_VA) || defined(HAVE_VA_INTEL)
+            static const char *predefined_vaapi_devices_list[] {"/dev/dri/renderD128",
```

Comment:
If we have all this code here, why do we have it in the sample app as well?

Reply:
this is "default device selection" part - for use-cases  when we do not want to choose suitable GPU device for Source and just select the first one. It's according to the "simplest sample" configuration, which is not interesting case as sample, as for me :) 

Please, check out comment in sample: https://github.com/opencv/opencv/pull/22212/files#diff-b4563bcdbe86eb5c8f1f2cce227809b02980777b25f5f4be2bb717826bedbebbR414

The current "complicated" sample demonstrate how to assing user-selected device for GAPI-pipelien: source, preproc, ie. The sample demonstrates this extended API usage using first-device in sample as "custom device". Therefore this "device selection part" is mostly the same in here and sample